installer/dev/install.cpp

```diff
@@ -318,14 +362,36 @@ namespace WindowsAppRuntimeInstaller
 
         PWSTR message{};
         if (FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM,
-                          nullptr, hr, 0, reinterpret_cast<PWSTR>(&message), 0, nullptr) == 0)
+            nullptr, hr, 0, reinterpret_cast<PWSTR>(&message), 0, nullptr) != 0)
         {
-            std::wcout << "Error " << hr << " (0x" << std::hex << hr << ") in FormatMessage()" << std::endl;
+            std::wcout << message;
         }
-        else
+
+        if (g_deploymentErrorExtendedHresult &&
+            (g_installStage == InstallStage::AddPackage ||
+                g_installStage == InstallStage::RegisterPackage))
         {
-            std::wcout << message;
-            LocalFree(message);
+            std::wcout << "ExtendedError: 0x" << std::hex << g_deploymentErrorExtendedHresult << " ";
+
+            if (FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM,
+                nullptr, g_deploymentErrorExtendedHresult, 0, reinterpret_cast<PWSTR>(&message), 0, nullptr) != 0)
+            {
+                std::wcout << message;
+            }
         }
+
+        if (g_deploymentErrorText && wcslen(g_deploymentErrorText) != 0 &&
```

Comment:
if `g_deploymentErrorText` is a string this would be `if (!g_deploymentErrorText.empty())`

See other commentss